app/gui/src/controller/searcher/component/builder.rs

```diff
@@ -125,13 +151,13 @@ impl List {
         }
     }
 
-    /// Build the list, sorting all group lists and groups' contents appropriately. Set the
-    /// favorites in the list keeping their order as passed.
-    pub fn build_with_favorites(self, favorites: component::group::List) -> component::List {
+    /// Build the list, sorting all group lists and groups' contents appropriately. (Does not sort
+    /// the [`component::List::favorites`].)
+    pub fn build(self) -> component::List {
         for group in self.module_groups.values() {
-            group.content.update_sorting("");
+            group.content.update_sorting_and_visibility("");
             if let Some(flattened) = &group.flattened_content {
-                flattened.update_sorting("");
+                flattened.update_sorting_and_visibility("");
             }
         }
```

Comment:
You may add a comment explaining, why favorites does _not_ need to be updated.

Reply:
As we agreed on Discord, it's enough that the method's docs already mention that favorites will *not* be updated by the method:
```
    /// ... (Does not sort
    /// the [`component::List::favorites`].)
```